eZ/Publish/Core/Repository/Repository.php

```diff
@@ -260,21 +269,28 @@ public function __construct(
             $this->serviceSettings['language']['languages'] = $this->serviceSettings['languages'];
         }
 
-        if ($user !== null) {
-            $this->setCurrentUser($user);
+        if ($user instanceof User) {
+            $this->currentUser = $user;
+            $this->currentUserRef = new UserReference($user->getUserId());
```

Comment:
Ok got what you mean now, but if someone uses `getCurrentUserReference`, and sends it around thinking it is just a user ref, putting it in session and stuff, then it better be just a user ref and not a 300 pound gorilla user object ;)

So I disagree :)